tools/mk_vendor

```diff
@@ -0,0 +1,854 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+NAME="mk_vendor"
+VERSION="0.1"
+
+USAGE="Usage:
+    ${NAME} make <configuration> [vendor-options]
+    ${NAME} locate <configuration> [vendor-options]
+    ${NAME} -h | --help
+    ${NAME} --version
+
+build and interact with vendor libraries
+
+Commands:
+    Get help for commands with ${NAME} COMMAND --help
+
+    make    build the vendor library
+    locate  find a vendor library
+
+Common Options:
+    -h, --help  display help and exit
+    --version   display version and exit
+
+Examples:
+    build the default version of OpenSSL:
+      \$ ${NAME} make openssl
+
+    build OpenSSL 1.1.1u:
```

Comment:
specify this is a preset? E.g.,:
> build OpenSSL 1.1.1u (as defined by the `preset` openssl111u):

Reply:
I am a bit unsure about this.

In my opinion, the help text is not there to document the tool, but rather provide a quick reminder of the commands and options. I feel like if we start talking about presets here, we should explain what it is in more details, and that the `openssl` part matches the file `puts/vendor/configs/openssl.toml`, and that ...

I think I would rather develop a more complete documentation separated from the help text.

Reply:
OK if this is documented elsewhere.